src/services/user.ts

```diff
@@ -114,6 +118,46 @@ export class UserService implements IUserService {
     return this.signup(login, password);
   }
 
+  public async requestResetPassword(login: string): Promise<IUserResetPassword> {
+    this.validateEmail(login);
+    const userDb = await User.findOne({ login }).exec();
+    if (userDb) {
+      const secret = JWT_SECRET + userDb.passwordHash;
+      const paylod = {
+        email: login,
+        id: userDb._id,
+      };
+      const token = jwt.sign(paylod, secret, { expiresIn: '15m' });
```

Comment:
This is quite aggressive. Let's increase that to 1 hour and make sure to let the user know in the email that "This link will expire in 1 hour"